src/common/state/redis.cc

```diff
@@ -1523,16 +1523,32 @@ void redis_actor_notification_table_subscribe_callback(redisAsyncContext *c,
     ActorID actor_id;
     WorkerID driver_id;
     DBClientID local_scheduler_id;
-    CHECK(sizeof(actor_id) + sizeof(driver_id) + sizeof(local_scheduler_id) ==
+    bool reconstruct;
+    CHECK(sizeof(actor_id) + sizeof(driver_id) + sizeof(local_scheduler_id) +
+              1 ==
           payload->len);
+    /* Parse the actor ID. */
     memcpy(&actor_id, payload->str, sizeof(actor_id));
+    /* Parse the driver ID. */
     memcpy(&driver_id, payload->str + sizeof(actor_id), sizeof(driver_id));
+    /* Parse the local scheduler ID. */
     memcpy(&local_scheduler_id,
            payload->str + sizeof(actor_id) + sizeof(driver_id),
            sizeof(local_scheduler_id));
+    /* Parse the reconstruct bit. */
+    if (*(payload->str + sizeof(actor_id) + sizeof(driver_id) +
```

Comment:
create a variable for (payload->str + sizeof(actor_id) + sizeof(driver_id) + sizeof(local_scheduler_id)) to avoid code duplication.